test/array-product/index.js

```diff
@@ -0,0 +1,58 @@
+var test = require('../util/test')(__filename);
+var product = require('../../packages/array-product');
+
+test('products', function(t) {
+  t.plan(5);
+
+  t.deepEqual(
+    product([]),
+    []
+  );
+
+  t.deepEqual(
+    product([[]]),
+    []
+  );
+
+  t.deepEqual(
+    product([['a']]),
+    ['a']
+  );
+
+  t.deepEqual(
+    product([['a', 'b'], [1, 2]]),
+    [['a', 1], ['a', 2], ['b', 1], ['b', 2]]
+  );
+
+  t.deepEqual(
+    product([['a', 'b'], [1, 2], [[], {}]]),
+    [
+      ['a', 1, []], ['a', 1, {}], ['a', 2, []], ['a', 2, {}],
+      ['b', 1, []], ['b', 1, {}], ['b', 2, []], ['b', 2, {}],
+    ]
+  );
+
```

Comment:
Can you add a test for arrays of different lengths e.g.` (['a', 'b', 'c'], [1, 2])`